ci/generate-workflows.py

```diff
@@ -54,19 +58,22 @@ def __init__(self, name, action, params=None, env=None, condition=None):
         self.env = env
         self.condition = condition
 
-    def render(self, f, env):
-        f.write(f"    - name: {yv(self.name)}\n")
-        f.write(f"      uses: {self.action}\n")
+    def render(self, f, env, depth=0):
+        indent = "  " * depth
+        f.write(f"{indent}- name: {yv(self.name)}\n")
+        f.write(f"{indent}  uses: {self.action}\n")
         if self.condition:
-            f.write(f"      if: {self.condition}\n")
+            f.write(f"{indent}  if: {self.condition}\n")
         if self.params:
-            f.write("      with:\n")
+            f.write(f"{indent}  with:\n")
             for k, v in self.params.items():
-                f.write(f"         {k}: {yv(v)}\n")
+                d = depth + 4 if v is str and "\n" in v else depth + 3
+                f.write(f"{indent}    {k}: {yv(v, d)}\n")
```

Comment:
I feel like this should just be:

```suggestion
                f.write(f"{indent}    {k}: {yv(v, depth + 1)}\n")
```

and then `yv` would do the appropriate indent based on the depth

Reply:
Yes, it is possible to use a simpler expression here, thanks! The value is `depth + 3`:

```
- name: "Cache cargo"             <--- depth "0"
  uses: actions/cache@v2.1.6      <--- depth +1
  with:                           <--- depth +1
    path: |                       <--- depth +2
      ~/.cargo/registry           <--- depth +3
      ~/.cargo/git                <--- depth +3
      target                      <--- depth +3
```